Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp

```diff
@@ -107,15 +107,37 @@ bool RenderSVGResourceMasker::applyResource(RenderElement& renderer, const Rende
 
 bool RenderSVGResourceMasker::drawContentIntoMaskImage(MaskerData* maskerData, const DestinationColorSpace& colorSpace, RenderObject* object)
 {
-    GraphicsContext& maskImageContext = maskerData->maskImage->context();
+    auto& maskImageContext = maskerData->maskImage->context();
+    auto objectBoundingBox = object->objectBoundingBox();
+
+    if (!drawContentIntoContext(maskImageContext, objectBoundingBox))
+        return false;
+
+#if !USE(CG)
+    maskerData->maskImage->transformToColorSpace(colorSpace);
+#else
+    UNUSED_PARAM(colorSpace);
+#endif
+
+    // Create the luminance mask.
+    if (style().svgStyle().maskType() == MaskType::Luminance)
+        maskerData->maskImage->convertToLuminanceMask();
+
+    return true;
+}
 
+bool RenderSVGResourceMasker::drawContentIntoContext(GraphicsContext& context, const FloatRect& objectBoundingBox, float effectiveZoom)
+{
     // Eventually adjust the mask image context according to the target objectBoundingBox.
     AffineTransform maskContentTransformation;
+
     if (maskElement().maskContentUnits() == SVGUnitTypes::SVG_UNIT_TYPE_OBJECTBOUNDINGBOX) {
-        FloatRect objectBoundingBox = object->objectBoundingBox();
         maskContentTransformation.translate(objectBoundingBox.location());
         maskContentTransformation.scale(objectBoundingBox.size());
-        maskImageContext.concatCTM(maskContentTransformation);
+        context.concatCTM(maskContentTransformation);
+    } else if (effectiveZoom != 1) {
```

Comment:
Are there tests with zooming?

Reply:
I removed this part since this function is called with effectiveZoom = 1. Zooming is handled when drawing the SVG resource by scaling the destination rectangle.